website/docs/API.md

```diff
@@ -515,20 +516,88 @@ Defined as:
 ```jsx
 function waitFor<T>(
   expectation: () => T,
-  { timeout: number = 4500, interval: number = 50 }
+  { timeout: number = 1000, interval: number = 50 }
 ): Promise<T> {}
 ```
 
-Waits for non-deterministic periods of time until your element appears or times out. `waitFor` periodically calls `expectation` every `interval` milliseconds to determine whether the element appeared or not.
+Waits for a period of time for the `expectation` callback to pass. `waitFor` may run the callback a number of times until timeout is reached, as specified by the `timeout` and `interval` options. The callback must throw an error when the expectation is not met. Returning any value, including a falsy one, will be treated as meeting the expectation, and the callback result will be returned to the caller of `waitFor` function.
```

Comment:
Not sure about the wording "waits ...for the callback to pass", is it the term to use in our use case? I don't have a much better solution though, maybe:
```suggestion
Waits for a period of time for the `expectation` callback to run without throwing. `waitFor` may run the callback a number of times until timeout is reached, as specified by the `timeout` and `interval` options. The callback must throw an error when the expectation is not met. Returning any value, including a falsy one, will be treated as meeting the expectation, and the callback result will be returned to the caller of `waitFor` function.
```

Reply:
It is specified in the same paragraph that the callback must throw an error and it is also further detailed with an example so I don't think there is a need for more precision here, wdyt ?

Reply:
agreed, it was not meant to add precision, i was just not sure about the use of the verb "pass" (I doubted my english programming vocabulary ^^) but overall yes it's very clear :) we can leave it as is